core/logger/audit/audit_logger.go

```diff
@@ -0,0 +1,312 @@
+package audit
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+
+	"github.com/pkg/errors"
+)
+
+const bufferCapacity = 2048
+const webRequestTimeout = 10
+
+type Data = map[string]any
+
+type AuditLogger interface {
+	services.ServiceCtx
+
+	Audit(eventID EventID, data Data)
+}
+
+type AuditLoggerConfig struct {
+	ForwardToUrl   *string
+	Environment    *string
+	JsonWrapperKey *string
+	Headers        []ServiceHeader
+}
+
+type HTTPAuditLoggerInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func NewAuditLoggerConfig(forwardToUrl string, isDev bool, jsonWrapperKey string, encodedHeaders string) (AuditLoggerConfig, error) {
+	if forwardToUrl == "" {
+		return AuditLoggerConfig{}, errors.Errorf("No forwardToURL provided")
```

Comment:
No point in using `errors.Errorf` if there is nothing to interpolate.
```suggestion
		return AuditLoggerConfig{}, errors.New("No forwardToURL provided")
```